File: sling-react-components/src/main/kotlin/com/cathcart93/sling/components/Application.kt
package com.cathcart93.sling.components

import com.cathcart93.sling.core.IReactController
import com.cathcart93.sling.core.ReactController
import com.cathcart93.sling.core.ReactProp

@ReactController(componentName = "Application")
class Application : IReactController {
    @ReactProp
    var content: Container? = null

    @ReactProp
    var isEditMode: Boolean = false
}

File: sling-react-js/src/main/js/components/application.js
import React from 'react';
import TreeContainer from './tree-container.js'
import components from './'
import wrapComponents from './../authoring-components/hoc/wrap-components.js'

const Application = ({ tree: { content, isEditMode } }) => <div>
    {/* TODO: It should be implemented in another way!, */}
    <TreeContainer tree={content} components={isEditMode ? wrapComponents(components) : components} />
</div>

export default Application;